src/com/trilead/ssh2/util/TimeoutService.java

```diff
@@ -20,25 +21,33 @@
  * @version $Id: TimeoutService.java,v 1.1 2007/10/15 12:49:57 cplattne Exp $
  */
 public class TimeoutService {
+    
+    
+    private  ScheduledFuture<?> scheduledFuture;
+    private final String hostname;
+    private final ThreadFactory threadFactory = new ThreadFactory() {
 
-    private final static ThreadFactory threadFactory = new ThreadFactory() {
-
-        private AtomicInteger count = new AtomicInteger();
-
+        @Override
         public Thread newThread(Runnable r) {
-            int threadNumber = count.incrementAndGet();
-            String threadName = "TimeoutService-" + threadNumber;
+            
+            String threadName = "Trilead_TimeoutService_"+hostname;
```

Comment:
To avoid possible conflicts between connections to the same host

```suggestion
            int threadNumber = count.incrementAndGet();
            String threadName = "Trilead_TimeoutService_" + hostname + "-" + threadNumber;
```